cabal-install/Distribution/Client/Utils/Parsec.hs

```diff
@@ -0,0 +1,65 @@
+module Distribution.Client.Utils.Parsec (
+    renderParseError,
+    ) where
+
+import Distribution.Client.Compat.Prelude
+import Prelude ()
+
+import qualified Data.ByteString       as BS
+import qualified Data.ByteString.Char8 as BS8
+
+import Distribution.Parsec.Common
+       (PError (..), PWarning (..), Position (..), showPError, showPWarning, zeroPos)
+import Distribution.Simple.Utils  (fromUTF8BS)
+
+-- | Render parse error highlighting the part of the input file.
+renderParseError
+    :: FilePath
+    -> BS.ByteString
+    -> [PError]
+    -> [PWarning]
+    -> String
+renderParseError filepath contents errors warnings = unlines $
+    [ "Errors encountered when parsing cabal file " <> filepath <> ":"
+    ]
+    ++ renderedErrors
+    ++ renderedWarnings
+  where
+    -- lines of the input file.
+    ls = BS8.lines contents
```

Comment:
Rename `ls` to `fileLines`? More descriptive and doesn't push column count over 80 below where its used.

Reply:
Also add type annotation?